test/commands/deploy/metadata/quick.nut.ts

```diff
@@ -97,6 +97,26 @@ describe('deploy metadata quick NUTs', () => {
         exitCode: 0,
       });
       assert(deploy);
+      assert(deploy.result.id !== validation.result.id, 'deploy result ID should not be the validation ID');
+      await testkit.expect.filesToBeDeployed(['force-app/**/*'], ['force-app/test/**/*']);
+    });
+
+    it('should deploy previously validated deployment (poll)', async () => {
+      const validation = await testkit.execute<DeployResultJson>('project:deploy:validate', {
+        args: '--source-dir force-app',
+        json: true,
+        exitCode: 0,
+      });
+      assert(validation);
+      await testkit.expect.filesToBeDeployed(['force-app/**/*'], ['force-app/test/**/*']);
+
+      const deploy = await testkit.execute<DeployResultJson>('project:deploy:quick', {
+        args: `--job-id ${validation.result.id} --wait 20`,
+        json: true,
+        exitCode: 0,
+      });
+      assert(deploy);
+      assert(deploy.result.id !== validation.result.id, 'deploy result ID should not be the validation ID');
```

Comment:
This assertion verifies the correct behavior.  If you remove the fix from quick.ts these tests will fail.